src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -9,12 +8,33 @@ internal static class ServiceFeatureOnOffOptions
     {
         public const string OptionName = "ServiceFeaturesOnOff";
 
+        private const bool CSharpClosedFileDiagnosticsEnabledByDefault = false;
+        private const bool DefaultClosedFileDiagnosticsEnabledByDefault = true;
+
         /// <summary>
         /// this option is solely for performance. don't confused by option name. 
         /// this option doesn't mean we will show all diagnostics that belong to opened files when turned off,
         /// rather it means we will only show diagnostics that are cheap to calculate for small scope such as opened files.
         /// </summary>
-        public static readonly PerLanguageOption<bool> ClosedFileDiagnostic = new PerLanguageOption<bool>(
-            OptionName, "Closed File Diagnostic", defaultValue: true, perLanguageDefaults: ImmutableDictionary<string, bool>.Empty.Add(LanguageNames.CSharp, false));
+        public static readonly PerLanguageOption<int> ClosedFileDiagnostic = new PerLanguageOption<int>(OptionName, "Closed File Diagnostic", defaultValue: -1);
```

Comment:
I have changed this option to take an int value, with the default being -1 (unset). This allows us to avoid different per language defaults (IOption2 change).
Additionally, in future, when we want to turn full solution analysis back on by default, we should be able to just delete the above flag `CSharpClosedFileDiagnosticsEnabledByDefault ` and this will ensure that we pick up the new default, unless user has explicitly set this value in Tools Options.